python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -2257,3 +2257,28 @@ def test_convert_unsupported_type_error_message():
     expected_msg = 'Conversion failed for column diff with type timedelta64'
     with pytest.raises(pa.ArrowNotImplementedError, match=expected_msg):
         pa.Table.from_pandas(df)
+
+
+def test_from_pandas_keeps_column_ordering():
+    # ARROW-3766
+    df = pd.DataFrame({
+        'partition': [0, 0, 1, 1],
+        'arrays': [[0, 1, 2], [3, 4], None, None],
+        'strings': [None, None, 'a', 'b']
+    })
+
+    schema = pa.schema([
+        # ('DPRD_ID', pa.int64()),
```

Comment:
I'm a bit hesitant with this issue, because previously `schema` argument was only used to provide optional type information for columns (IMO incorrectly). With this PR this behavior changes.
For example if I uncomment this line the new implementation fails, but the previous passes. Users might rely on this "wrong" functionality (I suppose `Table.from_pandas` is widely used).

What do You think @wesm ?